test/jdk/java/nio/file/Files/CopyAndMove.java

```diff
@@ -131,10 +147,15 @@ static void checkBasicAttributes(BasicFileAttributes attrs1,
         if (!attrs1.isSymbolicLink()) {
             long time1 = attrs1.lastModifiedTime().to(TimeUnit.SECONDS);
             long time2 = attrs2.lastModifiedTime().to(TimeUnit.SECONDS);
-
-            if (time1 != time2) {
-                System.err.format("File time for %s is %s\n", attrs1.fileKey(), attrs1.lastModifiedTime());
-                System.err.format("File time for %s is %s\n", attrs2.fileKey(), attrs2.lastModifiedTime());
+            long delta = Math.abs(Math.subtractExact(time1, time2));
+
+            // FAT32 volumes have a time stamp resolution of 2 seconds for
+            // last modified time (write time)
+            if ((delta != 0 && !targetVolumeIsFAT32) || delta > 2) {
+                System.err.format("File time for %s is %s\n",
+                                  attrs1.fileKey(), attrs1.lastModifiedTime());
+                System.err.format("File time for %s is %s\n",
+                                  attrs2.fileKey(), attrs2.lastModifiedTime());
```

Comment:
Are you 150% sure this is reliable?

Reply:
According to [1],

"[...] the resolution of create time on FAT is 10 milliseconds, while write time has a resolution of 2 seconds and access time has a resolution of 1 day [...]"

where "write time" is "last modified time." The testing I have done has thus far borne this out.

[1] [File Times](https://learn.microsoft.com/en-us/windows/win32/sysinfo/file-times).